joomla/joomla-cms
# Modify installer plugin interface and rewrite installer plugins

This PR decouples html rendering from the installer plugins.
#### Summary of Changes

Move plugin calls out of the layout to the model with appropriate b/c.
Show/hide 'install web installer' message logic is now in populateState method
Check for no plugin installed/enabled is now in the view itself.
Updated the new plugins in 3.6: packageinstaller, folderinstaller and urlinstaller  to new structure and events.
Layout for **isis** and **hathor** templates updated for new structure.

**Deprecating**:
InstallerViewInstall::$showJedAndWebInstaller
onInstallerViewBeforeFirstTab event
onInstallerViewAfterLastTab event

**New Event**:
onInstallerFetchInstallTypes

**Removed Event**:
onInstallerAddInstallationTab (not in public release yet, so no b/c concerns)
#### Testing Instructions

Enable all installer plugins, you should be able to see the relevant install tabs in the install page.
Disabling a plugin should hide the tab from install page.
Disable all plugins should show such message with link to plugin manager.
Reordering installer plugins from plugin manager should affect order of the tabs.
Submit buttons for each install type should perform relevant installation task as earlier.
#### Known Issue

Plugins created with old (now deprecating) triggers that directly echo the output are currently always rendered first on isis template. Internal ordering is still obeyed. (e.g. Install from Web)
